routers/user/home.go

```diff
@@ -103,7 +103,12 @@ func Dashboard(ctx *middleware.Context) {
 	feeds := make([]*models.Action, 0, len(actions))
 	for _, act := range actions {
 		if act.IsPrivate {
-			if has, _ := models.HasAccess(ctx.User, &models.Repository{Id: act.RepoId, IsPrivate: true}, models.ACCESS_MODE_READ); !has {
+			repo := &models.Repository{Id: act.RepoId, IsPrivate: true}
+			// This prevents having to retrieve the repository for each action
+			if act.RepoUserName == ctx.User.LowerName {
+				repo.OwnerId = ctx.User.Id
+			}
+			if has, _ := models.HasAccess(ctx.User, repo, models.ACCESS_MODE_READ); !has {
```

Comment:
Why when current user is owner but still check access?

Should we use `else if` here?